NoSuchFieldException for com.fasterxml.jackson.databind.ObjectMapper

I have a class that I test containing `com.fasterxml.jackson.databind.ObjectMapper` to store json in the file.
The test suite looks like this:
```kotlin
@StressCTest
@ModelCheckingCTest
class ATest : VerifierState() {
    private val a = A(Files.createTempFile("storage", ".json"))
...
}

class A(...) {
        private val objectMapper = ObjectMapper()
}
```

When running tests, I get the following error:
```
java.lang.ExceptionInInitializerError
	<here it points to the ObjectMapper line>
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
...
Caused by: java.lang.IllegalStateException: Cannot transform class com.fasterxml.jackson.databind.ser.BeanPropertyWriter
	at org.jetbrains.kotlinx.lincheck.TransformationClassLoader.loadClass(TransformationClassLoader.java:137)
	at com.fasterxml.jackson.databind.ObjectMapper.<clinit>(ObjectMapper.java:350)
	... 43 more
Caused by: java.lang.IllegalStateException: Cannot transform class com.fasterxml.jackson.databind.ser.PropertyWriter
	at org.jetbrains.kotlinx.lincheck.TransformationClassLoader.loadClass(TransformationClassLoader.java:137)
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:635)
...
Caused by: java.lang.NoSuchFieldException
	at org.jetbrains.kotlinx.lincheck.strategy.managed.ManagedStrategyTransformerKt.findField(ManagedStrategyTransformer.kt:1422)
	at org.jetbrains.kotlinx.lincheck.strategy.managed.ManagedStrategyTransformerKt.isFinalField(ManagedStrategyTransformer.kt:1407)
	... 68 more
```

--- a/src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategyTransformer.kt
+++ b/src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategyTransformer.kt
@@ -1404,7 +1404,8 @@ private fun isFinalField(ownerInternal: String, fieldName: String): Boolean {
     }
     return try {
         val clazz = Class.forName(internalName.canonicalClassName)
-        findField(clazz, fieldName).modifiers and Modifier.FINAL == Modifier.FINAL
+        val field = findField(clazz, fieldName) ?: throw NoSuchFieldException("No $fieldName in ${clazz.name}")
+        field.modifiers and Modifier.FINAL == Modifier.FINAL
     } catch (e: ClassNotFoundException) {
         throw RuntimeException(e)
     } catch (e: NoSuchFieldException) {
@@ -1412,14 +1413,17 @@ private fun isFinalField(ownerInternal: String, fieldName: String): Boolean {
     }
 }
 
-private fun findField(clazz: Class<*>, fieldName: String): Field {
-    var clazz: Class<*>? = clazz
-    do {
-        val fields = clazz!!.declaredFields
-        for (field in fields) if (field.name == fieldName) return field
-        clazz = clazz.superclass
-    } while (clazz != null)
-    throw NoSuchFieldException()
+private fun findField(clazz: Class<*>?, fieldName: String): Field? {
+    if (clazz == null) return null
+    val fields = clazz.declaredFields
+    for (field in fields) if (field.name == fieldName) return field
+    // No field found in this class.
+    // Search in super class first, then in interfaces.
+    findField(clazz.superclass, fieldName)?.let { return it }
+    clazz.interfaces.forEach { iClass ->
+        findField(iClass, fieldName)?.let { return it }
+    }
+    return null
 }
 
 private fun String.isNotPrimitiveType() = startsWith("L") || startsWith("[")

--- a/src/jvm/test/org/jetbrains/kotlinx/lincheck/test/transformation/FieldInheritedFromInterfaceTest.java
+++ b/src/jvm/test/org/jetbrains/kotlinx/lincheck/test/transformation/FieldInheritedFromInterfaceTest.java
@@ -1,0 +1,41 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2021 JetBrains s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ *
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>
+ */
+
+package org.jetbrains.kotlinx.lincheck.test.transformation;
+
+import org.jetbrains.kotlinx.lincheck.LinChecker;
+import org.jetbrains.kotlinx.lincheck.annotations.Operation;
+import org.jetbrains.kotlinx.lincheck.strategy.managed.modelchecking.ModelCheckingCTest;
+import org.junit.Test;
+
+@ModelCheckingCTest(requireStateEquivalenceImplCheck = false, iterations = 1)
+public class FieldInheritedFromInterfaceTest implements InterfaceWithField {
+    @Operation
+    public int get() { return INTERFACE_CONSTANT.getValue(); }
+
+    @Test
+    public void test() {
+        new LinChecker(FieldInheritedFromInterfaceTest.class, null).check();
+    }
+}
+
+interface InterfaceWithField {
+    public static final ValueHolder INTERFACE_CONSTANT = new ValueHolder(6);
+}

